nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -42,32 +49,110 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
   os.write((char *)builder.GetBufferPointer(), builder.GetSize());
   os.close();
 }
-} // namespace
 
-namespace nntrainer {
+/**
+ * @brief tensorflow operational node representation. This class contains,
+ * information to build operation flatbuffer
+ *
+ */
+class TfOpNode {
+private:
+  std::vector<std::shared_ptr<Var_Grad>> inputs;
+  std::vector<std::shared_ptr<Var_Grad>> outputs;
+  std::vector<std::shared_ptr<Var_Grad>> weights;
+
+  tflite::BuiltinOperator ops_type;
+
+  /// retrieve this from export_to
+  flatbuffers::Offset<void> builtin_ops;
+  tflite::BuiltinOptions builtin_option_type;
+};
+
+using TfOpNodes = std::vector<TfOpNode>;
+
+/**
+ * @brief tensorflow operation index map, this class manages operation index
+ * mapping
+ *
+ */
+class TfOpIdxMap {
+public:
+  TfOpIdxMap(std::vector<TfOpNode> nodes){};
+
+private:
+  /**
+   * @brief Bidirectional Index map
+   *
+   * @tparam T type of a underyling value
+   */
+  template <typename T> class BidirectionalIndexMap {
+    std::unordered_map<T, unsigned int> data2index; /**< data -> index map */
+    std::vector<T> index2data;                      /**< index -> data map */
+  };
+
+  float empty_buffer[0]; /**< unintialized tensor points to this buffer */
+
+  BidirectionalIndexMap<float *> buffer_map; /**< underlying buffer map */
+  BidirectionalIndexMap<tflite::BuiltinOperator> opcode_map; /**< opcode map */
+  BidirectionalIndexMap<Var_Grad *> variable_map;            /**< tensor map */
+};
+
+TfOpNodes
+buildOpNodes(std::shared_ptr<const GraphRepresentation> representation) {
+  /** NYI! */
+  return TfOpNodes();
+}
+
+flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<tflite::Buffer>>>
+buildBuffers(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
+  /** NYI! */
+  return flatbuffers::Offset<
+    flatbuffers::Vector<flatbuffers::Offset<tflite::Buffer>>>();
+}
+
+flatbuffers::Offset<
+  flatbuffers::Vector<flatbuffers::Offset<tflite::OperatorCode>>>
+buildOperatorCodes(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
+  /** NYI! */
+  return flatbuffers::Offset<
+    flatbuffers::Vector<flatbuffers::Offset<tflite::OperatorCode>>>();
+};
+
+flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<tflite::SubGraph>>>
+buildSubGraph(const TfOpNodes &nodes, const TfOpIdxMap &map,
```

Comment:
I wonder what is the SubGraph?

Reply:
subGraph is a concept from tflite file. It is basically same as our graph.